v1.5.0 changelogs (#55)
meilisearch/engine-team: meilisearch-release-changelog/v1.5.0.md

# v1.5.0 release changelogs

Meilisearch v1.5 introduces improvements in indexing speed and the possibility of triggering snapshots on demand.

🧰 All official Meilisearch integrations (including SDKs, clients, and other tools) are compatible with this Meilisearch release. Integration deployment happens between 4 to 48 hours after a new version becomes available.

Some SDKs might not include all new features—consult the project repository for detailed information. Is a feature you need missing from your chosen SDK? Create an issue letting us know you need it, or, for open-source karma points, open a PR implementing it (we'll love you for that ❤️).

# New features and improvements 🔥

## Indexing speed improvements

v1.5 improves indexing speed for text-heavy datasets. Datasets with fields containing many words (100 or more) should see a reduction of 5% to 20% to indexing times, with gains proportional to the amount of words in a document.

> This might result in minor impact to search result relevancy for queries containing 4 words or more. [Contact us in our GitHub Discussions page](https://github.com/orgs/meilisearch/discussions/704) if you have any issues or unexpected behavior regarding the relevancy.

Reviewer: ```suggestion
This might result in minor impact to search result relevancy for queries containing 4 words or more. [Contact us in our GitHub Discussions page](https://github.com/orgs/meilisearch/discussions/704) if this is significantly affecting your application.
```

> Indexing speed improvements might not be visible in datasets with fewer words (less than 20 words per fields), regardless of how many fields they contain.

Reviewer: ```suggestion
Indexing speed improvements might not be visible in datasets with fewer than 20 words per field, regardless of how many fields each document contains.
```

🧰 All official Meilisearch integrations (including SDKs, clients, and other tools) are compatible with this Meilisearch release. Integration deployment happens between 4 to 48 hours after a new version becomes available.

<!-- The following lines should NOT be put in the PRE-release changelogs -->
Some SDKs might not include all new features—consult the project repository for detailed information. Is a feature you need missing from your chosen SDK? Create an issue letting us know you need it, or, for open-source karma points, open a PR implementing it (we'll love you for that ❤️).

# New features and improvements 🔥

## Indexing speed improvements

For datasets containing a lot of words (and/or with a large variety of words), we improved the indexing speed between 5% and 20%, depending on your use cases and your documents. The more your documents contain a lot of words, the more the improvements will be visible.

The relevancy can be slightly different for queries containing 4 words or more. Let us know [here](https://github.com/meilisearch/product/discussions) if you have any issues or unexpected behavior.

> For datasets containing only a few words per field, even if you have a huge number of fields, the improvement might not be visible. Don't worry, we keep working on the indexing speed in the next sprint to address other use cases.

Author: > Do we have some sort of indicative number of what would be a lot of words for Meilisearch? A field with 10, 100, 1000 words? I think it would help a lot and make it easier for users to judge whether this is relevant to them.

To be 100% true, it's not a matter of number of words per field, but a number of different words in the datasets. So let's say at least a field with 100 words in their document, if it can help the users to have a number

> Also, should we already have a discussion prepared for this? Feels a bit weird to say "if you have problems/feedback, go to this page" when the linked page is a list of discussions unrelated to these performance improvements.

We do not want feedback about perf improvement, but about relevancy changes.
I created a product discussion -> https://github.com/orgs/meilisearch/discussions/704

> For datasets containing only a few words per field, even if you have a huge number of fields, the improvement might not be visible. Don't worry, we keep working on the indexing speed in the next sprint to address other use cases.

Author: I updated @guimachiavelli, let me know

> For datasets containing only a few words per field, even if you have a huge number of fields, the improvement might not be visible. Don't worry, we keep working on the indexing speed in the next sprint to address other use cases.

Reviewer: made a few minor edits to your changes, but we should be good to go
